docs/reference/installation/_interfaces/kubectl.cue

```diff
@@ -0,0 +1,134 @@
+package metadata
+
+installation: _interfaces: kubectl: {
+	title:       "Kubectl"
+	description: """
+		The [Kubernetes command-line tool](\(urls.kubectl)), kubectl, allows
+		users to run commands against Kubernetes clusters facilitating in
+		application deployment, scaling, monitoring, and introspection.
+		"""
+
+	archs: ["x86_64", "ARM64"]
+	paths: {
+		bin:         null
+		bin_in_path: null
+		config:      null
+	}
+	platform_name: installation.platforms.kubernetes.name
+
+	roles: [Name=string]: {
+		commands: {
+			_resource_type: string
+			_role:          string
+			install:        "kubectl apply -k ."
+			logs:           #"kubectl logs -n vector \#(_resource_type)/vector-\#(_role)"#
```

Comment:
If we ever need to generalize this, `vector-\#(_role)` can become `\#(_controller_resource_name)`, and the `\#(_resource_type)` can be `\#(_controller_resource_type)`.

In this case, the controller is something that maintains the `ReplicaSet`s, which in turn takes care of creating and deleting `Pod`s. It can be `DeamonSet`, `StatefulSet`, `Deployment`, etc.